src/main/java/org/fogbowcloud/manager/core/intercomponent/xmpp/handlers/RemoteNotifyEventHandler.java

```diff
@@ -25,25 +25,40 @@ public RemoteNotifyEventHandler() {
     @Override
     public IQ handle(IQ iq) {
         LOGGER.info("Received request for order: " + iq.getID());
-        Element queryElement = iq.getElement().element(IqElement.QUERY.toString());
-        Element orderElement = queryElement.element(IqElement.ORDER.toString());
-        String orderJsonStr = orderElement.getText();
-
-        Element orderClassNameElement = queryElement.element(IqElement.ORDER_CLASS_NAME.toString());
-        String className = orderClassNameElement.getText();
+        String orderJsonStr = unMarshallOrder(iq);
+        String className = unMarshallClassName(iq);
 
         IQ response = IQ.createResultIQ(iq);
 
         Gson gson = new Gson();
         Order order = null;
         try {
             order = (Order) gson.fromJson(orderJsonStr, Class.forName(className));
-            Element eventElement = queryElement.element(IqElement.EVENT.toString());
+            Element eventElement = marshallEvent(iq);
             Event event = gson.fromJson(eventElement.getText(), Event.class);
             RemoteFacade.getInstance().handleRemoteEvent(event, order);
         } catch (Exception e) {
             XmppExceptionToErrorConditionTranslator.updateErrorCondition(response, e);
         }
         return response;
     }
+
+	private Element marshallEvent(IQ iq) {
```

Comment:
I think it is not ok to have marshal (and unmarshal) to transform from/to IQ to Element, since both are transport objects. we should use it from/to business logic objects <--> json,xml etc